nix/ghc927.nix

```diff
@@ -0,0 +1,46 @@
+# To use this package set in your `haskell-flake` projects, set the
+# `basePackages` option as follows:
+#
+# > basePackages = config.haskellProjects.ghc927.outputs.finalPackages;
+#
+{ self, lib, ... }:
+
+let
+  # A function that enables us to write `foo = [ dontCheck ]` instead of `foo =
+  # lib.pipe super.foo [ dontCheck ]` in haskell-flake's `overrides`.
+  compilePipe = f: self: super:
+    lib.mapAttrs
+      (name: value:
+        if lib.isList value then
+          lib.pipe super.${name} value
+        else
+          value
+      )
+      (f self super);
+in
+{
+  perSystem = { pkgs, lib, config, ... }: {
+    haskellProjects.ghc927 = {
+      # This is not a local project, so disable those options.
+      packages = { };
+      devShell.enable = false;
+      autoWire = [ ];
+
+      # Uses GHC-9.2.7 package set as base
+      basePackages = pkgs.haskell.packages.ghc927;
```

Comment:
I think you should use `pkgs.haskellPackages` (already uses 9.2.x). It is cached by cache.nixos.org, unlike the versioned package sets.

Reply:
It was more of a _"err on the side of caution"_ kind of thing. There's also a more general `haskellPackages.92` set available, should we want to look at it.

Upgrading all these repos previously across multiple 9.2.x snapshots i found that even minor bumps in the stackage resolvers broke certain project builds (we depend on _alot_ of projects). Since nixpkgs somewhat tracks stackage snapshots, i think it's better to manually increment even minor point compiler and package-set versions after local testing and QA, generally being sure that nothing major breaks, with an explicit upgrade.

Would it be possible to cache these on our cachix services?

Reply:
Okay, fair enough. Can you add this as a code comment (as to why we are not using `pkgs.haskellPackages`)?

We do use cachix, but we are also hitting the space limit soon. I want us to self-host https://github.com/zhaofengli/attic at some point.